records.py

```diff
@@ -165,6 +165,11 @@ def dataset(self):
         # Create a new Tablib Dataset.
         data = tablib.Dataset()
 
+        # If the RecordCollection is empty, just return the empty set
+        # Check number of rows by typecasting to list
+        if len(list(self)) == 0:
```

Comment:
Also, an empty set directly represents that the table is empty right? Logically, if there isn't any data, throwing some sort of random error may be counter-intuitive.